### DISCUSSION TITLE
playwright

### DISCUSSION THREAD

**Discussion by markchen0304** [flet-dev/flet#4225] [Packages] [answered]
Please  add playwright to flet

**Comment by ndonkoHenri** [comment]
Playwright is mainly used to drive real browser engines by downloading their full binaries (hundreds of MB each) and launching them as child processes, then talking to them over a local websocket. I dont think it is possible to launch desktop binaries on mobile.
Can you share whats your usecase?

**Comment by Nwokike** [comment]
Bro use primp for whatever you want to do with playwright

**Comment by ndonkoHenri** [comment, marked as answer]
Playwright can't work in a Flet mobile app: the Python package is a thin client that spawns a bundled ~120 MB Node.js binary as a child process and drives full desktop browser engines (Chromium/Firefox/WebKit) over a local socket. Mobile app sandboxes can't launch arbitrary native executables, there's no Node runtime on device, and there's no source distribution to build from anyway.

As @Nwokike noted, if your use case is HTTP scraping / browser-like requests, [primp](https://pypi.flet.dev/) (browser-impersonating HTTP client) is available and works on mobile. For full browser automation/rendering, though, there's no mobile path. If you can share your exact use case we could discuss on the best alternative.
